internals/cli/masker/stream.go

```diff
@@ -0,0 +1,122 @@
+package masker
+
+import (
+	"bytes"
+	"io"
+	"sync"
+)
+
+// stream is a buffered io.Writer that masks all secrets written on it using a multipleMatcher.
+type stream struct {
+	dest          io.Writer
+	buf           IndexedBuffer
+	registerFrame func(*stream, int)
+
+	matcher     *multipleMatcher
+	matches     Matches
+	matchesLock sync.Mutex
+}
+
+// Write implements the io.Writer interface for the stream.
+// The written frame is stored in the buffer and it is registered in the Masker to make sure it is flushed from
+// the buffer after the buffer delay has passed.
+// The bytes are also passed to the secret matcher to check for any matches with secrets.
+func (s *stream) Write(p []byte) (int, error) {
```

Comment:
Are bytes (or frames) always buffered for the same amount of time (the constant buffer delay)? If so, maybe it's good to mention that?

Reply:
Added `constant`.